core/src/autogluon/core/ray/manager.py

```diff
@@ -0,0 +1,20 @@
+from autogluon.core.utils.try_import import try_import_ray
+
+class RayManager:
```

Comment:
Why will we need this singleton?

Reply:
Actually, why not just rely on `ray.init()`?

Reply:
Multiple places could use ray. For example, tabular bagging and hpo are both using ray. Having a singleton here to manage ray easier.

Reply:
I feel that this is not how a Singleton pattern is usually implemented: https://www.geeksforgeeks.org/singleton-pattern-in-python-a-complete-guide/amp/ . It is used when we need to protect certain resources. However, what’s stored here is the module “ray”, which itself can be viewed as a singleton.

Reply:
Also, does ray provide methods for accessing the config, e.g., something like ray._config?

Reply:
So you can't really implement singleton in python because there's no private constructor...
The singleton is mainly here to maintain the config not ray module.

Reply:
If that’s the case, we may try to check if there are other ways to access the ray config and return that with a function. This singleton implementation looks unnecessary to me because it wraps a module.

Reply:
The reason I wrap a module is so that we can do lazy import.

Reply:
How about doing that with a function? Ray should have stored the configuration somewhere.

Reply:
Yea, if ray have that info, then a function makes sense